ksqldb-engine/src/main/java/io/confluent/ksql/query/QueryRegistryImpl.java

```diff
@@ -290,7 +292,9 @@ public PersistentQueryMetadata createOrReplacePersistentQuery(
       );
     } else {
       query = queryBuilder.buildPersistentQueryInDedicatedRuntime(
-          ksqlConfig,
+          oldQuery != null && sharedRuntimeId.isPresent()
```

Comment:
nit: I might compute this config outside of the function call.  

As a separate question, is the logic around keeping and/or overriding the properties tested below?

Reply:
looks like we are still figuring out how we want to handle the configs. There has been more discussion since I made this change. I am just going to remove this logic and only have the location control in this PR and handle the config changes in a follow up once we are set on what we want to do